src/main.ts

```diff
@@ -0,0 +1,99 @@
+import type { OutputWriter } from './output-writer/index.js';
+import {
+  combineOutputWriters,
+  createGoogleSheetsOutputWriter,
+  createCSVOutputWriter,
+} from './output-writer/index.js';
+
+import * as path from 'path';
+import type { CrawlOptions } from './crawl.js';
+import { crawl } from './crawl.js';
+import { initWorkerThreads } from './lighthouse.js';
+import type { ModifiedConsole } from './cli.js';
+
+/**
+ * Creates output writers for each specified output format,
+ * and returns a single merged output writer that updates each of them individually.
+ */
+const getOutputWriter = async (outputs: string[]): Promise<OutputWriter> => {
+  const outputWriters = await Promise.all(
+    outputs.map((output) => {
+      if (output === 'google-sheets') return createGoogleSheetsOutputWriter();
+      const ext = path.extname(output);
+      if (ext === '.csv') return createCSVOutputWriter(output);
+      throw new Error(
+        `Invalid output format: ${ext} (${output}). Expected <filename>.csv, or google-sheets`
+      );
+    })
+  );
+
+  return combineOutputWriters(outputWriters);
+};
+
+export const enum State {
+  Pending,
+  ReportInProgress,
+  ReportSuccess,
+  ReportFailure,
```

Comment:
Should this be `ReportPending` to match the format of the others?
```suggestion
  ReportPending,
  ReportInProgress,
  ReportSuccess,
  ReportFailure,
```

Reply:
Maybe I should just take Report off of it... or change it to Lighthouse. The bottom three represent the state of lighthouse, but the first one means the URL hasn't been given to lighthouse yet.

Reply:
Ah, I see. Hmm...maybe remove "Report" and leave them as `Pending`, `InProgress`, `Success`, `Failure` then as you suggested is fine.